include/envoy/network/transport_socket.h

```diff
@@ -181,6 +182,11 @@ class TransportSocketFactory {
    */
   virtual bool implementsSecureTransport() const PURE;
 
+  /**
+   * @return CertificateValidationContextConfig the certificate validation context config.
+   */
+  virtual const Ssl::CertificateValidationContextConfig* certificateValidationContext() const PURE;
```

Comment:
We need a bit of abstraction for this by not leaking `Ssl` namespace into here, perhaps a Protobuf::Message& config() or abstract class `TransportSocketConfig`. `Ssl::Connection` in `TransportSocket` is due to historical reason and that is to be removed.

Reply:
It looks like ATLS is not using certificate validation context. How do we tell whether ATLS config is meant for mutual TLS?

Reply:
did you mean ALTS? If you expose TransportSocket proto does that work?